From 6b354a4e19c158dd29e0ac278ee39a5575bd3b6a Mon Sep 17 00:00:00 2001
From: Navin Kochar <navin.kochar@intel.com>
Date: Tue, 20 Aug 2013 14:06:31 +0530
Subject: Disabling BT Clean Turn On and modifying BT Off Cleanup sequence.

To send SW RF Kill Cmd, hci interface should be up.
So, clean up was done after calling SW RF KILL.

Change-Id: I4064399ae9a9ef6874ba7a4fd036122a199d1da8
Signed-off-by: Navin Kochar <navin.kochar@intel.com>
---
 main/bte_main.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/main/bte_main.c b/main/bte_main.c
index 62c6b6b..19c990b 100644
--- a/main/bte_main.c
+++ b/main/bte_main.c
@@ -41,6 +41,10 @@
 **  Constants & Macros
 *******************************************************************************/
 
+#ifndef BT_CLEAN_TURN_ON_DISABLED
+#define BT_CLEAN_TURN_ON_DISABLED TRUE
+#endif
+
 /* Run-time configuration file */
 #ifndef BTE_STACK_CONF_FILE
 #define BTE_STACK_CONF_FILE "/etc/bluetooth/bt_stack.conf"
@@ -319,8 +323,8 @@ static void bte_hci_disable(void)
 
     if (bt_hc_if)
     {
-        bt_hc_if->cleanup();
         bt_hc_if->set_power(BT_HC_CHIP_PWR_OFF);
+	bt_hc_if->cleanup();
         if (hci_logging_enabled == TRUE ||  hci_logging_config == TRUE)
             bt_hc_if->logging(BT_HC_LOGGING_OFF, hci_logfile);
     }
-- 
1.7.4.1

